Add IngressMTLS policy support (#1166)
nginxinc/kubernetes-ingress: docs-web/configuration/policy-resource.md

     - The JWT policy configures NGINX Plus to authenticate client requests using JSON Web Tokens.
     - `jwt <#jwt>`_
     - No*
   * - ``IngressMTLS``

>      - The IngressMTLS policy controls client verification.

Reviewer: ```suggestion
     - The IngressMTLS policy configures client certificate verification.
```

```
In this example the Ingress Controller will use the configuration from the first policy reference `jwt-policy-one`, and ignores `jwt-policy-two`.

### IngressMTLS

> The IngressMTLS policy controls client validation.

Reviewer: ```suggestion
The IngressMTLS policy configures client certificate verification.
```

> For example, the following policy will verify a client using the CA certificate specified in the `ingress-mtls-secret`:

Reviewer: ```suggestion
For example, the following policy will verify a client certificate using the CA certificate specified in the `ingress-mtls-secret`:
```

```yaml
ingressMTLS:
  clientCertSecret: ingress-mtls-secret

>   verifyClient: "on"

Reviewer: ```suggestion
  verifyClient: on
```

  verifyDepth: 1
```

> 

Reviewer: We need to capture that the policy requires TLS termination enabled and that the policy should be referenced only in the spec.

How about adding a paragraph here:

---

A VirtualServer that references an IngressMTLS policy must:
* Enable [TLS termination](https://deploy-preview-1166--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#virtualserver-tls).
* Reference the policy in the VirtualServer [`spec`](https://deploy-preview-1166--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#virtualserver-specification). It is not allowed to reference an IngressMTLS policy in a [`route `](https://deploy-preview-1166--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#virtualserver-route) or in a VirtualServerRoute [`subroute`](https://deploy-preview-1166--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#virtualserverroute-subroute).

If the conditions above are not met, NGINX will send `500` error response to clients.

> Note: The feature is implemented using the NGINX [ngx_http_ssl_module](https://nginx.org/en/docs/http/ngx_http_ssl_module.html).

```eval_rst
.. list-table::
   :header-rows: 1

   * - Field
     - Description
     - Type
     - Required
   * - ``clientCertSecret``
     - The name of the Kubernetes secret that stores the CA certificate. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``ca.crt``, otherwise the secret will be rejected as invalid.
     - ``string``
     - Yes
   * - ``verifyClient``

>      - Verification for the client. Possible values are ``on``, ``off``, ``optional``, ``optional_no_ca``

Reviewer: ```suggestion
     - Verification for the client. Possible values are ``on``, ``off``, ``optional``, ``optional_no_ca``. The default is ``on``.
```

     - ``string``
     - No
   * - ``verifyDepth``

>      - Sets the verification depth in the client certificates chain.

Reviewer: ```suggestion
     - Sets the verification depth in the client certificates chain. The default is ``1``.
```

     - ``int``
     - No
```

#### IngressMTLS Merging Behavior

> A VirtualServer/VirtualServerRoute can reference multiple IngressMTLS policies. However, only one can be applied. Every subsequent reference will be ignored. For example, here we reference two policies:

Reviewer: ```suggestion
A VirtualServer can reference only a single IngressMTLS policy. Every subsequent reference will be ignored. For example, here we reference two policies:
```
